PollBuddy-Server/backend/routes/users.js

```diff
@@ -661,6 +663,226 @@ router.post("/me/groups", function (req, res) {
+ * @getdata {void} None
+ * @postdata {void} None
+ * @returns {void} Status 405 { "result": "failure", "error": "Route not availible."}
+ * @name backend/users/forgotpassword_GET
+ * @param {string} path - Express path
+ * @param {callback} callback - function handler for route
+ */
+router.get("/forgotpassword/",function (req,res) {
+  return res.status(405).send(createResponse(null,"Route not availible."));
+});
+
+/**
+ * This route is not used. It is simply there to have some response to /api/users/forgotpassword/ when using POST
+ * @getdata {void} None
+ * @postdata {void} None
+ * @returns {void} Status 405 { "result": "failure", "error": "Route not availible."}
+ * @name backend/users/forgotpassword_POST
+ * @param {string} path - Express path
+ * @param {callback} callback - function handler for route
+ */
+router.post("/forgotpassword/",function (req,res) {
+  return res.status(405).send(createResponse(null,"Route not availible."));
+});
+
+/**
+ * This route is not used. It is simply there to have some response to /api/users/forgotpassword/submit when using GET
+ * @getdata {void} None
+ * @postdata {void} None
+ * @returns {void} Status 405 { "result": "failure", "error": "GET not availible.use POST"}
+ * @name backend/users/forgotpassword/submit_GET
+ * @param {string} path - Express path
+ * @param {callback} callback - function handler for route
+ */
+router.get("/forgotpassword/submit",function (req,res) {
+  return res.status(405).send(createResponse(null,"GET not availible.use POST"));
+});
+
+/**
+ * The route primes a given user's data so their password may be reset
+ * accepts either the users username or email
+ * sets in the user's data a random 32char code and an expiration date 1 hour from the requests handling
+ * @getdata {void} None
+ * @postdata {void} email : String , username : String
+ * @returns {void} on success : Status 200
+ * On failure : Status 500 { "result": "failure", "error": "neither username nor email provided"}
+ *              Status 500 { "result": "failure", "error": "could not update user"}
+ *              Status 500 { "result": "failure", "error": "could not find user"}
+ * @name backend/users/forgotpassword/submit_POST
+ * @param {string} path - Express path
+ * @param {callback} callback - function handler for route
+ */
+router.post("/forgotpassword/submit/",function (req,res) {
+  let email = req.body.email;
+  const username = req.body.username;
+
+  let document;
+
+  if(email){
+    email = email.toLowerCase();
+    document = mongoConnection.getDB().collection("users").findOne({"Email":email});
+  }else if(username){
+    document = mongoConnection.getDB().collection("users").findOne({"UserName":username});
+  }else{
+    return res.status(500).send(createResponse("Neither username nor email provided.", null));
+  }
+
+  document
+    .then(
+      result => {
+        if(result) {
+          const alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
```

Comment:
Just to avoid some potential confusion if people type these for whatever reason, can you use this alphabet? `ABCDEFGHKLMNPQRSTUVWXYZabcdefghkmnpqrstuvwxyz23456789`